### PR TITLE
:sparkles: Added Webhook related endpoints

### PR DESCRIPTION
**Pull request by trag1c**
`Channel.get_webhooks`
`Guild.get_webhooks`
`WebhookCompatibility` enum
`Webhook.edit`
`Webhook.delete`
`Webhook.execute`
`Webhook.execute_github`
`Webhook.execute_slack`
`Webhook.get_message`
`Webhook.delete_mesage`
`Webhook.edit_message`
`Webhook.from_id`

 ### Check off the following

-   [x] I have tested my changes with the current requirements
-   [x] My Code follows the pep8 code style.

### CODE REVIEW

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pincer/objects/guild/channel.py [hunk 1]
```diff
@@ -344,6 +351,24 @@ async def send(self, message: Union[Embed, Message, str]) -> UserMessage:
...
+        """|coro|
+        Get all webhooks in the channel.
+        Requires the ``MANAGE_WEBHOOKS`` permission.
+
+        Yields
+        -------
+        AsyncGenerator[:class:`~.pincer.objects.guild.webhook.Webhook`, None]
+        """
+        data = await self._http.get(f"channels/{self.id}/webhooks")
+        for webhook_data in data:
```
Comment (Lunarmagpie) [inline pincer/objects/guild/channel.py:364]:
```suggestion
        for webhook_data in await self._http.get(f"channels/{self.id}/webhooks"):
```

File: pincer/objects/guild/guild.py [hunk 2]
```diff
@@ -1532,6 +1533,21 @@ async def delete_template(
...
+        """|coro|
+        Returns an async generator of the guild webhooks.
+
+        Yields
+        -------
+        AsyncGenerator[:class:`~pincer.objects.guild.webhook.Webhook`, None]
+            The guild webhook object.
+        """
+        data = await self._http.get(f"guilds/{self.id}/webhooks")
+        for webhook_data in data:
```
Comment (Lunarmagpie) [inline pincer/objects/guild/guild.py:1546]:
```suggestion
        for webhook_data in await self._http.get(f"guilds/{self.id}/webhooks"):
```

File: pincer/objects/guild/webhook.py [hunk 3]
```diff
@@ -90,3 +101,342 @@ class Webhook(APIObject):
...
+
+        data = await self._http.patch(
+            request_route,
+            data=request_data
+        )
+        return Webhook.from_dict(
+            construct_client_dict(self._client, data)
+        )
+
+    async def delete(self, token: Optional[str] = None) -> None:
```
Comment (Lunarmagpie) [inline pincer/objects/guild/webhook.py:150]: `-> None` isn't needed

File: pincer/objects/guild/webhook.py [hunk 4]
```diff
@@ -90,3 +101,342 @@ class Webhook(APIObject):
...
     guild_id: APINullable[Optional[Snowflake]] = MISSING
+
+    async def edit(
+        self,
+        *,
+        name: Optional[str] = None,
+        avatar: Optional[str] = None,
+        channel_id: Optional[Snowflake] = None,
+        token: Optional[str] = None
+    ) -> None:
```
Comment (Lunarmagpie) [inline pincer/objects/guild/webhook.py:113]:
```suggestion
    ) -> Webhook:
```

File: pincer/objects/guild/webhook.py [hunk 5]
```diff
@@ -90,3 +101,342 @@ class Webhook(APIObject):
...
+                    + (f"?{thread_id=}" if thread_id else "")
+                )
+            )
+        )
+
+    async def delete_message(
+        self,
+        message_id: Snowflake,
+        thread_id: Snowflake
+    ) -> None:
```
Comment (Lunarmagpie) [inline pincer/objects/guild/webhook.py:333]: `-> None` isn't needed

File: pincer/objects/guild/channel.py (hunk 1 above)
Comment (trag1c) [reply pincer/objects/guild/channel.py:364]: Too long

File: pincer/objects/guild/webhook.py [hunk 6]
```diff
@@ -90,3 +102,341 @@ class Webhook(APIObject):
...
+        """
+        await self._http.delete(
+            f"webhooks/{self.id}/{self.token}/messages/{message_id}"
+            + (f"?{thread_id=}" if thread_id else "")
+        )
+
+    @overload
+    async def edit_message(
+        self,
+        message_id: Snowflake,
```
Comment (Lunarmagpie) [inline pincer/objects/guild/webhook.py:352]: Are you able to use `self.id` for this?

File: pincer/objects/guild/webhook.py [hunk 7]
```diff
@@ -90,3 +102,341 @@ class Webhook(APIObject):
...
+            (multipart/form-data only)
+        attachments: Optional[List[:class:`~pincer.objects.message.attachment.Attachment`]]
+            Attached files to keep and
+            possible descriptions for new files
+        """
+        ...
+
+    async def edit_message(
+        self,
+        message_id: Snowflake,
```
Comment (Lunarmagpie) [inline pincer/objects/guild/webhook.py:393]: Are you able to use `self.id` for this?

**Review by Lunarmagpie — CHANGES_REQUESTED** [review]

File: pincer/objects/guild/webhook.py (hunk 6 above)
Comment (trag1c) [reply pincer/objects/guild/webhook.py:352]: Yeah

File: pincer/objects/guild/webhook.py (hunk 7 above)
Comment (Lunarmagpie) [reply pincer/objects/guild/webhook.py:393]: resolving cause i did the same thing twice lol

File: pincer/objects/guild/webhook.py [hunk 8]
```diff
@@ -4,21 +4,33 @@
...
     from ..user.user import User
     from ..guild.guild import Guild
     from ..guild.channel import Channel
     from ...utils.types import APINullable
     from ...utils.snowflake import Snowflake
+    from ...client import Client
 
+class WebhookCompatibility(Enum):
+    GitHub = "/github"
+    Slack = "/slack"
```
[the remark this answers is not in the document]
Comment (trag1c) [reply pincer/objects/guild/webhook.py:32]: Yes

File: pincer/exceptions.py [hunk 9]
```diff
@@ -185,6 +185,10 @@ def from_desc(cls, _type: str, max_size: int, cur_size: int):
         )
 
 
+class EmbedOverflow(PincerError):
+    """Exception that is raised when too many embeds are passed in."""
+
+
```
Comment (Endercheif) [inline pincer/exceptions.py:191]: Add `EmbedOverflow` to `__init__.py`

**Review by Endercheif — CHANGES_REQUESTED** [review]

**Review by Lunarmagpie — APPROVED** [review]

**Review by Sigmanificient — APPROVED** [review]

**Review by Endercheif — APPROVED** [review]
<3

**Pull request merged**